chalk-rust-ir/src/lib.rs

```diff
@@ -87,6 +87,10 @@ pub struct StructFlags {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
+/// A rust intermediate representation (rust_ir) of a Trait Definition.
+///
+/// Not to be confused with a rust_ir for a Trait Implementation, which is
+/// represented with ??? JRL
 pub struct TraitDatum<I: Interner> {
     pub id: TraitId<I>,
```

Comment:
Doc-comment:

The id of this trait; could be used to load the trait datum by invoking the [`trait_datum`] method.

[`trait_datum`]: http://rust-lang.github.io/chalk/chalk_solve/trait.RustIrDatabase.html#tymethod.trait_datum

Reply:
The doc for `RustIrDatabase::trait_datum` says that this fn returns the trait datum for an impl, it seems like these comments are contradictory.

Reply:
yeah that's a copy and paste bug in the doc comment for `trait_datum`, good catch :)